### DISCUSSION TITLE
ADIF-Import and -Export of all ADIF-Fields in actual specification

### DISCUSSION THREAD

**Discussion by dg9vh** [magicbug/Cloudlog#1593] [Ideas / Feature Requests]
Hi,
for those not sure what I am talking about now, visit the page: http://www.adif.org.uk/
Here you find the actual definition of adif standard and it's fields. Actually when importing adif-files into cloudlog only a subset of all delivered data would be imported, so we lose data we could import. So I think we should have the following 4 steps on the timeline to implement:

- import of all standard-adif-fields delivered by the imported ADIF-File or via API-calls
- storing them in the QSO-Database
- making them available for editing in the Web-UI
- exporting them when exporting ADIF-Files (not the specialized exports for eQSL, LoTW and so on but when using ADIF-Export-function in 
![image](https://user-images.githubusercontent.com/13950650/188229038-146aaf96-85ec-47a7-a699-792282afea97.png)

This would give us absolutely a good flexibility to have a central point of logging for different sources (like wsjt-x, fldigi and so on...) and not losing any data by throwing information away because of "we cannot import it because we can't store it actually".

How do you all think about it? I think importing, storing and exporting should be implemented first (because this is "background-stuff") and in a second step we should implement the editing in the webUI.

This means that the logging-window (live&past-logging) would persist, as it is in a first step but the editing windows whould be getting much richer.

How about?

73 de Kim
DG9VH

**Comment by AndreasK79** [comment]
I second this @dg9vh. 

The biggest task would be to implement editing of everything.
Adding fields, importing and exporting should be much quicker to implement.

Do you know which fields are missing?

**Reply by dg9vh** [reply]
It's the following fields missing in our database:

**Field Name:**
AWARD_SUBMITTED
GUEST_OP
IOTA_ISLAND_ID
MY_ARRL_SECT
MY_POSTAL_CODE_INTL
QSLMSG_INTL
VE_PROV

GUEST_OP and VE_PROV are specified as import only so here we can supersede them with the actual representation mentioned in the specification.

I did not check, if all fields we have in the database are set in the importer/exporter-function. This would be our first step to have them imported and exported as is to not lose data via importing and managing them in Cloudlog.

For the editing of all fields missing we could have an "expert editing" tab that could be switched on/off by configuring it in the user-profile maybe? Here only those fields not already regularly within the edit-windows should be added to avoid dublicates and confusion.

**Reply by dg9vh** [reply]
I toke a look yesterday at the code but did not get through it in the short time... I saw, there are some libs in usage (Adif_Helper and so on)... I think, this is one place to add the missing things beneath adding them into the database...
